packages/api/lambdas/sf-semaphore-down.js

```diff
@@ -62,34 +71,39 @@ async function decrementQueueSemaphore(queueName) {
 }
 
 /**
- * Filter workflow messages from SNS to prepare array of semaphore decrement tasks.
+ * Handle Cloudwatch event and decrement semaphore, if necessary.
  *
- * @param {Object} event - incoming message from SNS
- * @returns {Array<Promise>} - Array of promises for semaphore decrement operations
+ * @param {Object} event - incoming event from Cloudwatch
+ * @returns {Promise}
  */
-function getSemaphoreDecrementTasks(event) {
-  return get(event, 'Records', [])
-    // Skip if this record is not from SNS or if the SNS message is empty
-    .filter((record) => has(record, 'Sns.Message') && !isNil(record.Sns.Message))
-    .map((record) => JSON.parse(record.Sns.Message))
-    .filter((message) => isDecrementMessage(message))
-    .map((message) => getQueueName(message))
-    .map((queueName) => decrementQueueSemaphore(queueName));
+async function handleSemaphoreDecrementTask(event) {
+  if (!isSfExecutionEvent(event)) {
```

Comment:
I don't love have two separate early returns (one here and below), but it seemed unnecessary to proceed further if the event is not for a Step Function. 

I separately don't love having to do `return Promise.resolve`, but ESlint requires it

Reply:
This function doesn't need to return anything, so you could just say `if (...) return` and then `await decrementQueueSemaphore()` at the end.